core/server/master/src/main/java/alluxio/web/MasterWebServer.java

```diff
@@ -97,7 +98,7 @@ public void init() throws ServletException {
         errorHandler.addErrorPage(404, "/");
         mServletContextHandler.setErrorHandler(errorHandler);
       }
-    } catch (MalformedURLException e) {
+    } catch (IOException e) {
```

Comment:
justification for this change:

Here is the implementation of the code that previously throwed MalformedURLException

  public static Resource newResource(String resource, boolean useCaches) throws IOException {
    URL url = null;

    try {
      url = new URL(resource);
    } catch (MalformedURLException var6) {
      if (!resource.startsWith("ftp:") && !resource.startsWith("file:") && !resource.startsWith("jar:")) {
        try {
          if (resource.startsWith("./")) {
            resource = resource.substring(2);
          }

          File file = (new File(resource)).getCanonicalFile();
          return new PathResource(file);
        } catch (IOException var5) {
          var5.addSuppressed(var6);
          throw var5;
        }
      }

      LOG.warn("Bad Resource: " + resource, new Object[0]);
      throw var6;
    }

Reply:
Are you saying now this code will throw other IOExceptions other than `MalformedURLException`?